Kernel/WorkQueue.cpp

```diff
@@ -28,6 +29,11 @@ UNMAP_AFTER_INIT WorkQueue::WorkQueue(StringView name)
     if (name_kstring.is_error())
         TODO();
     (void)Process::create_kernel_process(thread, name_kstring.release_value(), [this] {
+        // NOTE: We need to disable interrupts, because somewhere in this function Spinlock.unlock
+        //       is called. For x86_64 the default for kernel threads/process is to start with interrupts disabled,
+        //       but for aarch64 the default is to start a threads with interrupts enabled (which seems way more logical anyway).
+        Processor::disable_interrupts();
```

Comment:
This looks fishy. We call Spinlock.unlock() after retrieving an item from `m_items`, as it's SpinLockProtected.

It's also possible that the wait_on method of the wait_queue does things with interrupts. 

This should probably be a FIXME, as explicitly disabling interrupts at the start of thread loop function feels very awkward.

If we get around to making spin locks less about interrupts and more about locking out CPUs, then we'll want to remove this.

Reply:
I agree it's a bit fishy, and I don't completely understand why this code needs to be executed without interrupts. I added an ifdef for aarch64, as we only need to disable interrupts for aarch64, and I also changed the NOTE to a FIXME, and also changed the wording of the FIXME. Let me know what you think!